src/views/product-integration/components/header/index.tsx

```diff
@@ -24,12 +27,26 @@ export default function Header({
 	versions,
 	hideVersions,
 	description,
+	activeRelease,
 }: HeaderProps) {
+	// /waypoint/integrations/docker[/0.10.2]
+	const { asPath } = useRouter()
+
+	// derive some values from the path
+	const productSlug = asPath.split('/')[1]
+	const integrationSlugAndVersion = asPath.split('/').slice(3).join('/')
+	const [integrationSlug, _] = integrationSlugAndVersion.split('/')
```

Comment:
I don't think this component should be coupled to the router.

The data that you're pulling out of the URL structure should be able to be passed to this header as props, which will keep it independent of where it is rendered in the app.

Reply:
I'll look at generating the links, server side — that might be the most reliable

Reply:
Done serverside: [b681eed](https://github.com/hashicorp/dev-portal/pull/1429/commits/b681eedc96c70776311d40a3c70748fabcf1e68e)

Reply:
I like your resolution here!